oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java

```diff
@@ -0,0 +1,236 @@
+import ch.qos.logback.core.spi.FilterReply;
+
+/**
+ * Tests the index augmentation feature.
+ */
+public class DynamicBoostTest extends AbstractQueryTest {
+    
+    public static final String ASSET_NODE_TYPE = 
+            "[dam:Asset]\n" + 
+            " - * (UNDEFINED) multiple\n" + 
+            " - * (UNDEFINED)\n" + 
+            " + * (nt:base) = oak:TestNode VERSION";
+    
+    private static final String UNSTRUCTURED = "nt:unstructured";
+    
+    private final SimpleIndexAugmentorFactory factory = new SimpleIndexAugmentorFactory();
+
+    @Override
+    protected void createTestIndexNode() throws Exception {
+        setTraversalEnabled(false);
+    }
+
+    @Override
+    protected ContentRepository createRepository() {
+        IndexTracker tracker = new IndexTracker();
+        LuceneIndexEditorProvider editorProvider = new LuceneIndexEditorProvider(null,
+            new ExtractedTextCache(0, 0),
+            factory, Mounts.defaultMountInfoProvider());
+        LuceneIndexProvider provider = new LuceneIndexProvider(tracker,
+            factory);
+        return new Oak()
+            .with(new OpenSecurityProvider())
+            .with((QueryIndexProvider) provider)
+            .with((Observer) provider)
+            .with(editorProvider)
+            .createContentRepository();
+    }
+    
+    @Test public void withFieldProvider() throws Exception {
+        NodeTypeRegistry.register(root, toInputStream(ASSET_NODE_TYPE), "test nodeType");
+        createIndex("dam:Asset");
+        root.commit();
+
+        factory.indexFieldProvider = new IndexFieldProviderImpl();
+        
+        Tree t = createTestNodes(root);
+        
+        AsyncLogFilter filter = new AsyncLogFilter(IndexFieldProviderImpl.class);
+        ListAppender<ILoggingEvent> appender = filter.subscribe();
+        root.commit();
+        // this is not detected
+        updateTestNode(t, 20);
+        root.commit();
+        // now we change an indexed property: this is not detected with the IndexFieldProvider
+        t.getParent().setProperty("updateCount", 2);
+        updateTestNode(t, 30);
+        root.commit();
+        filter.unsubscribe(appender);
+        assertEquals("[[TRACE] Added augmented fields: jcr:content/metadata/predictedTags/[my, a, my:a], 10.0]", appender.list.toString());
+    }
+    
+    @Test public void withDynamicBoost() throws Exception {
+        NodeTypeRegistry.register(root, toInputStream(ASSET_NODE_TYPE), "test nodeType");
+        Tree props = createIndex("dam:Asset");
+        Tree pt = createNodeWithType(props, "predictedTags", UNSTRUCTURED);
+        pt.setProperty("name", "jcr:content/metadata/predictedTags/.*");
+        pt.setProperty("isRegexp", true);
+        pt.setProperty("dynamicBoost", true);
+        pt.setProperty("propertyIndex", true);
+        root.commit();
+        
+        Tree t = createTestNodes(root);
+
+        AsyncLogFilter filter = new AsyncLogFilter(LuceneDocumentMaker.class);
```

Comment:
There is a LogCustomizer in oak-commons to do log related verifications. Maybe we can use that here. That also makes use of ILoggingEvent.

Reply:
Oh I didn't know about LogCustomizer, I'll check.

Reply:
Used LogCustomizer now. This simplified the test case quite a bit!